aiida/backends/tests/engine/test_work_chain.py

```diff
@@ -294,977 +296,994 @@ def tearDown(self):
-
-    def test_incorrect_outline(self):
-
-        class Wf(WorkChain):
-
-            @classmethod
-            def define(cls, spec):
-                super(Wf, cls).define(spec)
-                # Try defining an invalid outline
-                spec.outline(5)
-
-        with self.assertRaises(TypeError):
-            Wf.spec()
-
-    def test_same_input_node(self):
-
-        class Wf(WorkChain):
-
-            @classmethod
-            def define(cls, spec):
-                super(Wf, cls).define(spec)
-                spec.input('a', valid_type=Int)
-                spec.input('b', valid_type=Int)
-                # Try defining an invalid outline
-                spec.outline(cls.check_a_b)
-
-            def check_a_b(self):
-                assert 'a' in self.inputs
-                assert 'b' in self.inputs
-
-        x = Int(1)
-        run_and_check_success(Wf, a=x, b=x)
-
-    def test_context(self):
-        A = Str("a")
-        B = Str("b")
-
-        test_case = self
-
-        class ReturnA(WorkChain):
-
-            @classmethod
-            def define(cls, spec):
-                super(ReturnA, cls).define(spec)
-                spec.outline(cls.result)
-
-            def result(self):
-                self.out('res', A)
-
-        class ReturnB(WorkChain):
-
-            @classmethod
-            def define(cls, spec):
-                super(ReturnB, cls).define(spec)
-                spec.outline(cls.result)
-
-            def result(self):
-                self.out('res', B)
-
-        class Wf(WorkChain):
-
-            @classmethod
-            def define(cls, spec):
-                super(Wf, cls).define(spec)
-                spec.outline(cls.s1, cls.s2, cls.s3)
+    def test_run_base_class(self):
+        """Verify that it is impossible to run, submit or instantiate a base `WorkChain` class."""
+        with self.assertRaises(exceptions.InvalidOperation):
+            WorkChain()
 
-            def s1(self):
-                return ToContext(r1=self.submit(ReturnA), r2=self.submit(ReturnB))
+        with self.assertRaises(exceptions.InvalidOperation):
+            launch.run(WorkChain)
 
-            def s2(self):
-                test_case.assertEquals(self.ctx.r1.out.res, A)
-                test_case.assertEquals(self.ctx.r2.out.res, B)
+        with self.assertRaises(exceptions.InvalidOperation):
+            launch.run_get_node(WorkChain)
 
-                # Try overwriting r1
-                return ToContext(r1=self.submit(ReturnB))
+        with self.assertRaises(exceptions.InvalidOperation):
+            launch.run_get_pid(WorkChain)
 
-            def s3(self):
-                test_case.assertEquals(self.ctx.r1.out.res, B)
-                test_case.assertEquals(self.ctx.r2.out.res, B)
+        with self.assertRaises(exceptions.InvalidOperation):
+            launch.submit(WorkChain)
 
-        run_and_check_success(Wf)
+#     def test_run(self):
```

Comment:
Is this huge number of lines replaced with comments expected?

Reply:
No, I forgot to uncomment, I will fix it

Reply:
Should be fixed now